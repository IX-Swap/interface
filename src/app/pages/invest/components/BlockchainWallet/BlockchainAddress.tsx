import { Box, Typography } from '@mui/material'
import { CHAIN_INFO } from 'config/blockchain/constants'
import { stringTruncate } from 'config/utils'
import { shortenAddress } from 'helpers/blockchain'
import { copyToClipboard } from 'helpers/clipboard'
import React from 'react'
import { useStyles } from './BlockchainWallet.styles'
export interface BlockchainAddressProps {
  account?: string | null
  chainId?: number
  labelName: string
}
export const BlockchainAddress = ({
  account,
  chainId,
  labelName
}: BlockchainAddressProps) => {
  const classes = useStyles()
  const info = CHAIN_INFO[chainId as number]
  return (
    <Box
      className={classes.addressBlock}
      onClick={() => copyToClipboard(account ?? '')}
    >
      <img src={info.logoUrl} alt={'Icon'} className={classes.chainLogo} />
      <Typography variant='subtitle2' fontWeight={600}>
        {labelName.length > 0 ? stringTruncate(labelName, 10) + ' : ' : ''}
        {shortenAddress(account ?? '')}
      </Typography>
    </Box>
  )
}
