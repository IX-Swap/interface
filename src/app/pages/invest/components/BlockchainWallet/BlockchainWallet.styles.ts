import makeStyles from '@mui/styles/makeStyles'

export const useStyles = makeStyles(theme => ({
  label: {
    color: theme.palette.mode === 'light' ? '#666666' : '#ffffff'
  },
  addressBlock: {
    color: theme.palette.primary.main,
    display: 'flex',
    gap: theme.spacing(1),
    alignItems: 'center',
    background: theme.palette.background.paper,
    border: `1px solid ${theme.palette.primary.main}`,
    borderRadius: theme.spacing(6),
    padding: theme.spacing(0.25, 1.5),
    textAlign: 'center',
    verticalAlign: 'middle'
  },
  chainLogo: {
    width: 17,
    height: 17
  }
}))
