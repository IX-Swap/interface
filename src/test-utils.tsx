import React from 'react'
import { Router } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import {
  createGenerateClassName,
  StylesProvider
} from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'
import Themes from './v2/themes'
import { history } from 'v2/history'
import { UserProvider } from 'v2/auth/context'
import { UserStore } from 'v2/auth/context/store'
import { PasswordResetProvider } from 'v2/auth/context/password-reset'
import PasswordResetStore from 'v2/auth/context/password-reset/store'
import { PasswordResetStep } from 'v2/auth/context/password-reset/types'
import { AuthorizerTableStore } from 'v2/app/pages/authorizer/context/store'
import { AuthorizerTableStoreProvider } from 'v2/app/pages/authorizer/context'
import { DepositStore } from 'v2/app/pages/accounts/pages/banks/context/store'
import { DepositStoreProvider } from 'v2/app/pages/accounts/pages/banks/context'
import { ServicesProvider } from 'v2/services/useServices'
import { renderHook, RenderHookResult } from '@testing-library/react-hooks'
import { BreadcrumbsProvider } from 'v2/hooks/useBreadcrumbs'

const generateClassName = createGenerateClassName({
  productionPrefix: 'ix'
})

const BaseProviders: React.FC = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={Themes.default}>
        <BreadcrumbsProvider>
          <Router history={history}>{children}</Router>
        </BreadcrumbsProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

const customRenderer = (
  ui: any,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => {
  return render(ui, { wrapper: BaseProviders, ...options })
}

export const testComponentId = 'test-component'

export const fakeUserStore: Partial<UserStore> = {
  activeTab: 0,
  setActiveTab: jest.fn()
}

export const renderWithUserStore = (
  ui: any,
  store?: Partial<UserStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <UserProvider value={{ ...fakeUserStore, ...store }}>
        {children}
      </UserProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakePasswordResetStore: Partial<PasswordResetStore> = {
  currentStep: PasswordResetStep.Request,
  setCurrentStep: jest.fn()
}

export const renderWithPasswordResetStore = (
  ui: any,
  store?: Partial<PasswordResetStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <PasswordResetProvider value={{ ...fakePasswordResetStore, ...store }}>
        {children}
      </PasswordResetProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeAuthorizerTableStore: Partial<AuthorizerTableStore> = {
  idKey: '__',
  uri: '/api/source',
  _getItemId: jest.fn()
}

export const renderWithAuthorizerTableStore = (
  ui: any,
  store?: Partial<AuthorizerTableStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <AuthorizerTableStoreProvider
        value={{ ...fakeAuthorizerTableStore, ...store }}
      >
        {children}
      </AuthorizerTableStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeDepositStore: Partial<DepositStore> = {}

export const renderWithDepositStore = (
  ui: any,
  store?: Partial<DepositStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <DepositStoreProvider value={{ ...fakeDepositStore, ...store }}>
        {children}
      </DepositStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const renderHookWithServiceProvider = (
  hookFn: any,
  store: object = {}
): RenderHookResult<any, any> => {
  const WithServiceProvider: React.FC = ({ children }) => (
    <ServicesProvider value={store}>{children}</ServicesProvider>
  )

  return renderHook(hookFn, { wrapper: WithServiceProvider })
}

export * from '@testing-library/react'
export { customRenderer as render }
