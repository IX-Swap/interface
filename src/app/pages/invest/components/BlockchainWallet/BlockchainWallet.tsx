import { Box, Button, Hidden, Typography } from '@mui/material'
import { useWithdrawalAddressAdded } from 'app/pages/accounts/pages/withdrawalAddresses/hooks/useWithdrawalAddressAdded'
import { WithdrawalAddressesRoute } from 'app/pages/accounts/pages/withdrawalAddresses/router/config'
import { useStyles } from 'app/pages/invest/components/BlockchainWallet/BlockchainWallet.styles'
import { WalletModalContext } from 'components/WalletModal/WalletModalContextWrapper'
import { isEmptyString } from 'helpers/strings'
import { useActiveWeb3React } from 'hooks/blockchain/web3'
import React, { useContext } from 'react'
import { Link } from 'react-router-dom'
import { BlockchainAddress } from './BlockchainAddress'
export const BlockchainWallet = () => {
  const classes = useStyles()
  const web3 = useActiveWeb3React()
  const context = useContext(WalletModalContext)
  const { account, chainId, active } = web3
  const { found, label } = useWithdrawalAddressAdded(account)
  return (
    <>
      <>
        {isEmptyString(account) || !active ? (
          <Button
            variant='contained'
            color='primary'
            onClick={context?.toggleModal}
          >
            Connect Wallet
          </Button>
        ) : (
          <>
            {!found && active && (
              <Button
                variant='contained'
                color='primary'
                component={Link}
                to={WithdrawalAddressesRoute.create}
              >
                Add Wallet
              </Button>
            )}
            {found && active && (
              <Box display='flex' gap={1}>
                <Hidden lgDown>
                  <Typography className={classes.label} variant='subtitle1'>
                    Your Wallet:
                  </Typography>
                </Hidden>
                <BlockchainAddress
                  chainId={chainId}
                  account={account}
                  labelName={label}
                />
              </Box>
            )}
          </>
        )}
      </>
    </>
  )
}
