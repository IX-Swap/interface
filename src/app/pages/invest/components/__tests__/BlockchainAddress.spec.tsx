import { BlockchainAddress } from 'app/pages/invest/components/BlockchainWallet/BlockchainAddress'
import React from 'react'
import { render } from 'test-utils'

describe('BlockchainAddress', () => {
  afterEach(async () => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const { container } = render(
      <BlockchainAddress
        account={'0x67ed490d810c41263758e7355cef720ffed68cbc'}
        chainId={1}
        labelName='wallet'
      />
    )
    expect(container).toMatchSnapshot()
  })
})
