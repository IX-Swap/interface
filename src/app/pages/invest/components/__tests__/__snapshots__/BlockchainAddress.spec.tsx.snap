// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockchainAddress should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-addressBlock-2 MuiBox-root css-0"
  >
    <img
      alt="Icon"
      class="makeStyles-chainLogo-3"
      src="image-mock"
    />
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 makeStyles-address-4 css-s0c2fe-MuiTypography-root"
    >
      wallet: 
      0x67ed...8cbc
    </h6>
  </div>
</div>
`;
